plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/webview/BrowserConnector.kt

```diff
@@ -598,6 +641,85 @@ class BrowserConnector(
         }
     }
 
+    fun parseFindingsMessages(messagesMap: Map<String, *>) {
+        try {
+            val additionalMessages = messagesMap["additionalMessages"] as? MutableList<Map<String, Any>>
+            val findingsMessages = additionalMessages?.filter { message ->
+                if (message.contains("messageId")) {
+                    (message["messageId"] as String).endsWith(CODE_REVIEW_FINDINGS_SUFFIX) ||
+                        (message["messageId"] as String).endsWith(DISPLAY_FINDINGS_SUFFIX)
+                } else {
+                    false
+                }
+            }
+            val scannedFiles = mutableListOf<VirtualFile>()
+            if (findingsMessages != null) {
+                for (findingsMessage in findingsMessages) {
+                    additionalMessages.remove(findingsMessage)
+                    val gson = Gson()
```

Comment:
nit: would recommend using the same instance of Gson throughout the class

Reply:
will add in follow up PR